src/libData/AccountData/AccountStoreSC.tpp

```diff
@@ -224,48 +228,61 @@ bool AccountStoreSC<MAP>::UpdateAccounts(const uint64_t& blockNum,
 
     // Undergo scilla runner
     bool ret = true;
-    std::string runnerPrint;
 
-    pid = -1;
-    auto func2 = [this, &runnerPrint, &ret, &pid, gasRemained,
-                  &receipt]() mutable -> void {
-      if (!SysCommand::ExecuteCmd(
-              SysCommand::WITH_OUTPUT_PID,
-              GetCreateContractCmdStr(m_root_w_version, gasRemained),
-              runnerPrint, pid)) {
-        LOG_GENERAL(WARNING, "ExecuteCmd failed: " << GetCreateContractCmdStr(
-                                 m_root_w_version, gasRemained));
-        receipt.AddError(EXECUTE_CMD_FAILED);
-        ret = false;
+    if (ret_checker) {
+      std::string runnerPrint;
+
+      pid = -1;
+      auto func2 = [this, &runnerPrint, &ret, &pid, gasRemained,
+                    &receipt]() mutable -> void {
+        if (!SysCommand::ExecuteCmd(
+                SysCommand::WITH_OUTPUT_PID,
+                GetCreateContractCmdStr(m_root_w_version, gasRemained),
+                runnerPrint, pid)) {
+          LOG_GENERAL(WARNING, "ExecuteCmd failed: " << GetCreateContractCmdStr(
+                                   m_root_w_version, gasRemained));
+          receipt.AddError(EXECUTE_CMD_FAILED);
+          ret = false;
+        }
+        cv_callContract.notify_all();
+      };
+      DetachedFunction(1, func2);
+
+      {
+        std::unique_lock<std::mutex> lk(m_MutexCVCallContract);
+        cv_callContract.wait(lk);
       }
-      cv_callContract.notify_all();
-    };
-    DetachedFunction(1, func2);
 
-    {
-      std::unique_lock<std::mutex> lk(m_MutexCVCallContract);
-      cv_callContract.wait(lk);
-    }
+      if (m_txnProcessTimeout) {
+        LOG_GENERAL(WARNING,
+                    "Txn processing timeout! Interrupt current contract "
+                    "deployment, pid: "
+                        << pid);
+        if (pid >= 0) {
+          kill(pid, SIGKILL);
+        }
+        receipt.AddError(EXECUTE_CMD_TIMEOUT);
+        ret = false;
+      }
 
-    if (m_txnProcessTimeout) {
-      LOG_GENERAL(WARNING,
-                  "Txn processing timeout! Interrupt current contract "
-                  "deployment, pid: "
-                      << pid);
-      if (pid >= 0) {
-        kill(pid, SIGKILL);
+      if (ret && !ParseCreateContract(gasRemained, runnerPrint, receipt)) {
+        ret = false;
       }
-      receipt.AddError(EXECUTE_CMD_TIMEOUT);
-      ret = false;
+      if (!ret) {
+        gasRemained =
+            std::min(transaction.GetGasLimit() -
+                         std::max((unsigned int)transaction.GetCode().size(),
+                                  CONTRACT_CREATE_GAS),
+                     gasRemained);
+      }
+    } else {
+      gasRemained =
```

Comment:
Since at this point we know that the checker failed and the runner was not run, this calculation for `gasRemained` can simply be 

```
assert(gasRemained == transaction.GetGasLimit()); // I'm not sure how asserts are used in the codebase. Remove if needed.
gasRemained = gasRemained - createGasLimit; // createGasLimit defined in my previous comment above.
```